acceptance-tests/src/main/kotlin/org/ostelco/at/common/StripeEventListener.kt

```diff
@@ -0,0 +1,167 @@
+                                    }
+                        }
+                    })
+
+    private fun waitForStripeEvent(topic: String, subscription: String, eventType: String, timeout: Long, cmp: (Event) -> Boolean): Boolean {
+        var status = false
+        val subscriber = PubSubSubscriber(
+                topic = topic,
+                subscription = subscription,
+                cmp = cmp
+        )
+
+        try {
+            runBlocking {
+                status = withTimeoutOrNull(timeout) {
+                    subscriber.start()
+                    while (!subscriber.eventSeen()) {
+                        delay(200L)
+                    }
+                    "Done"
+                } != null
+                if (!status)
+                    logger.error("Reached timeout limit $timeout while waiting for event $eventType")
+            }
+        } finally {
+            subscriber.stop()
+        }
+
+        return status
+    }
+}
+
+/**
+ * The pubsub setup is for use with emulator only.
+ */
+class PubSubSubscriber(val topic: String,
+                       val subscription: String,
+                       val cmp: (Event) -> Boolean) {
+
+    private val logger by getLogger()
+    private val gson = Gson()
+
+    private val hostport: String = System.getenv("PUBSUB_EMULATOR_HOST")
+    private val project: String = System.getenv("GCP_PROJECT_ID")
+
+    private val subscriptionName = ProjectSubscriptionName.of(project,
+            subscription)
+    private val channel = ManagedChannelBuilder.forTarget(hostport)
+            .usePlaintext()
+            .build()
+    private val channelProvider = FixedTransportChannelProvider
+            .create(GrpcTransportChannel.create(channel))
+    private val receiver = MessageReceiver { message, consumer ->
+        handler(message.data, consumer)
+    }
+
+    /* For managment. */
+    private var subscriber = Subscriber.newBuilder(subscriptionName, receiver)
+            .setChannelProvider(channelProvider)
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .build()
+
+    fun start() {
+        logger.info("ACCEPTANCE TEST PUBSUB: Enabling subscription ${subscription} for project ${project} and topic ${topic}")
+
+        try {
+            subscriber.startAsync().awaitRunning()
+        } catch (e: Exception) {
+            logger.error("ACCEPTANCE TEST PUBSUB: Failed to connect to service: {}",
+                    e.message)
+        }
+    }
+
+    fun stop() {
+        try {
+            subscriber.stopAsync()
+        } catch (e: Exception) {
+            logger.error("ACCEPTANCE TEST PUBSUB: Error disconnecting to service: {}",
+                    e.message)
+        }
+    }
+
+    private fun handler(message: ByteString, consumer: AckReplyConsumer) {
+        try {
+            gson.fromJson(message.toStringUtf8(), Event::class.java)
+                    .let {
+                        if (cmp(it)) eventSeen = true
```

Comment:
What does this mean? My initial reading is  "compare it with itself", which makes no sense.  A comment is in order.